src/main/minio.js

```diff
@@ -164,6 +165,23 @@ export class Client {
     this.reqOptions = {}
   }
 
+  static async getTempCredentialsFromJWTToken(minioURL,token,valid){
```

Comment:
This can be refactored using the built in request  like:
```
  handleHttpRequest (requestConfig)  {
    return new Promise((resolve, reject) => {
      const request = this.transport.request(requestConfig, res => {
        let chunks = []
        res.on('data', chunk => {
          chunks.push(chunk)
        })

        res.on('end', () => {
          let body = Buffer.concat(chunks).toString()
          resolve(body)
        })

      })
      //handling error
      request.on('error', (e) => {
        reject(e)
      })

      if(!_.isEmpty(requestConfig.data)){ //POST, DELETE,
        request.write(JSON.stringify(requestConfig.data))
      }
      request.end()
    })
  }
```

and can be called like:
```
const response = minioClient.handleHttpRequest({
  hostname:"sphprakash.free.beeceptor.com",
  protocol:"https:",
  method:'POST',
  data:{
    Action:"AssumeRoleWithWebIdentity",
    DurationSeconds:20,
    WebIdentityToken:encodeURIComponent("test-token"),
    Version:"2011-06-15"
  } 
})

response.then((data)=>{
  console.log("Response", data)
})
```